I'm resolving CI errors. For the reel board works fine, but for the cc3220sf_launchxl I have an error, that it can't find #define DB_VAL, and also some problem with the system call. I always used ARM reel board for development, so that kind of errors got me stuck. @ioannisg Do you know why for the ARM board cc3220sf_launchx such kind of errors during build? For the ARM reel board I can compile program.
```
 -MT CMakeFiles/app.dir/src/arm_syscalls.c.obj -MF CMakeFiles/app.dir/src/arm_syscalls.c.obj.d -o CMakeFiles/app.dir/src/arm_syscalls.c.obj   -c /home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c
In file included from /home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:274:
zephyr/include/generated/syscalls/test_arm_cpu_write_reg_mrsh.c: In function 'z_mrsh_test_arm_cpu_write_reg':
zephyr/include/generated/syscalls/test_arm_cpu_write_reg_mrsh.c:14:10: error: 'struct k_thread' has no member named 'syscall_frame'
   14 |  _current->syscall_frame = ssf;
      |          ^~
zephyr/include/generated/syscalls/test_arm_cpu_write_reg_mrsh.c:23:10: error: 'struct k_thread' has no member named 'syscall_frame'
   23 |  _current->syscall_frame = NULL;
      |          ^~
In file included from /home/maksim/zephyrproject/zephyr/subsys/testsuite/ztest/include/ztest.h:60,
                 from /home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:7:
/home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c: In function 'test_syscall_cpu_scrubs_regs':
/home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:299:37: error: 'DB_VAL' undeclared (first use in this function)
  299 |   zassert_not_equal(arm_reg_val[i], DB_VAL,
      |                                     ^~~~~~
/home/maksim/zephyrproject/zephyr/subsys/testsuite/ztest/include/ztest_assert.h:93:12: note: in definition of macro 'zassert'
   93 |  z_zassert(cond, msg ? ("(" default_msg ")") : (default_msg), \
      |            ^~~~
/home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:299:3: note: in expansion of macro 'zassert_not_equal'
  299 |   zassert_not_equal(arm_reg_val[i], DB_VAL,
      |   ^~~~~~~~~~~~~~~~~
/home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:299:37: note: each undeclared identifier is reported only once for each function it appears in
  299 |   zassert_not_equal(arm_reg_val[i], DB_VAL,
      |                                     ^~~~~~
/home/maksim/zephyrproject/zephyr/subsys/testsuite/ztest/include/ztest_assert.h:93:12: note: in definition of macro 'zassert'
   93 |  z_zassert(cond, msg ? ("(" default_msg ")") : (default_msg), \
      |            ^~~~
/home/maksim/zephyrproject/zephyr/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c:299:3: note: in expansion of macro 'zassert_not_equal'
  299 |   zassert_not_equal(arm_reg_val[i], DB_VAL,
      |   ^~~~~~~~~~~~~~~~~
[24/126] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
ninja: build stopped: subcommand failed.
```

--- a/tests/arch/arm/arm_thread_swap/README.txt
+++ b/tests/arch/arm/arm_thread_swap/README.txt
@@ -18,6 +18,7 @@ behaves as expected. In particular, the test verifies that:
   at thread swap-in
 - the mode variable (when building with support for either user
   space or FP shared registers) is saved and restored properly.
+- the CPU registers are scrubbed after system call
 
 Notes:
   The test verifies the correct behavior of the thread context-switch,

--- a/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c
+++ b/tests/arch/arm/arm_thread_swap/src/arm_syscalls.c
@@ -23,6 +23,7 @@
 #if defined(CONFIG_USERSPACE)
 
 #define PRIORITY 0
+#define DB_VAL 0xDEADBEEF
 
 static struct k_thread user_thread;
 static K_THREAD_STACK_DEFINE(user_thread_stack, 1024);
@@ -236,6 +237,64 @@ void test_arm_syscalls(void)
 		(uint32_t *)i, NULL, NULL,
 		K_PRIO_COOP(PRIORITY), K_USER,
 		K_NO_WAIT);
+}
+
+void z_impl_test_arm_cpu_write_reg(void)
+{
+	/* User thread CPU write registers system call for testing
+	 *
+	 * Verify the following
+	 * - Write 0xDEADBEEF values during system call into registers
+	 * - In main test we will read that registers to verify
+	 * that all of them were scrubbed and do not contain any sensitive data
+	 */
+
+	/* Part below is made to test that kernel scrubs CPU registers
+	 * after returning from the system call
+	 */
+	TC_PRINT("Writing 0xDEADBEEF values into registers\n");
+	__asm__ volatile (
+		"ldr r0, =0xDEADBEEF;\n\t"
+		"ldr r1, =0xDEADBEEF;\n\t"
+		"ldr r2, =0xDEADBEEF;\n\t"
+		"ldr r3, =0xDEADBEEF;\n\t"
+		);
+	TC_PRINT("Exit from system call\n");
+}
+
+static inline void z_vrfy_test_arm_cpu_write_reg(void)
+{
+	z_impl_test_arm_cpu_write_reg();
+}
+#include <syscalls/test_arm_cpu_write_reg_mrsh.c>
+
+/**
+ * @brief Test CPU scrubs registers after system call
+ *
+ * @details - Call from user mode a syscall test_arm_cpu_write_reg(),
+ * the system call function writes into registers 0xDEADBEEF value
+ * - Then in main test function below check registers values,
+ * if no 0xDEADBEEF value detected, that means CPU scrubbed registers
+ * before exit from the system call.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_syscall_cpu_scrubs_regs(void)
+{
+	uint32_t arm_reg_val[4];
+
+	test_arm_cpu_write_reg();
+
+	__asm__ volatile ("mov %0, r0" : "=r"(arm_reg_val[0]));
+	__asm__ volatile ("mov %0, r1" : "=r"(arm_reg_val[1]));
+	__asm__ volatile ("mov %0, r2" : "=r"(arm_reg_val[2]));
+	__asm__ volatile ("mov %0, r3" : "=r"(arm_reg_val[3]));
+
+	for (int i = 0; i < 4; i++) {
+		zassert_not_equal(arm_reg_val[i], DB_VAL,
+				"register value is 0xDEADBEEF, "
+				"not scrubbed after system call.");
+	}
 }
 #endif /* CONFIG_USERSPACE */
 /**

--- a/tests/arch/arm/arm_thread_swap/src/main.c
+++ b/tests/arch/arm/arm_thread_swap/src/main.c
@@ -8,6 +8,7 @@
 
 extern void test_arm_thread_swap(void);
 extern void test_arm_syscalls(void);
+extern void test_syscall_cpu_scrubs_regs(void);
 
 void test_main(void)
 {
@@ -16,7 +17,8 @@ void test_main(void)
 	ztest_run_test_suite(arm_thread_swap);
 #if defined(CONFIG_USERSPACE)
 	ztest_test_suite(arm_syscalls,
-		ztest_unit_test(test_arm_syscalls));
+		ztest_unit_test(test_arm_syscalls),
+		ztest_user_unit_test(test_syscall_cpu_scrubs_regs));
 	ztest_run_test_suite(arm_syscalls);
 #endif
 }

--- a/tests/arch/arm/arm_thread_swap/src/test_syscalls.h
+++ b/tests/arch/arm/arm_thread_swap/src/test_syscalls.h
@@ -9,6 +9,7 @@
 #include <zephyr.h>
 
 __syscall void test_arm_user_syscall(void);
+__syscall void test_arm_cpu_write_reg(void);
 
 #include <syscalls/test_syscalls.h>
 

